tutorials/preprocessing/plot_55_setting_eeg_reference.py

```diff
@@ -192,6 +192,7 @@
 # the source modeling is performed. In contrast, applying an average reference
 # by the traditional subtraction method offers no such guarantee.
 #
+# XXX: IS THIS TRUE?
 # For these reasons, when performing inverse imaging, *MNE-Python will
 # automatically average-reference the EEG channels if they are present and no
 # reference strategy has been specified*. If you want to perform inverse
```

Comment:
note: I am uncertain about how to change the lines above. Will MNE-Python or will it NOT apply auto average refs?

Reply:
As far as I can tell, it is *not* automatically applied:

- `_check_reference` is called in `_apply_inverse` [here](https://github.com/mne-tools/mne-python/blob/master/mne/minimum_norm/inverse.py#L885)
- `_check_reference` ([here](https://github.com/mne-tools/mne-python/blob/master/mne/minimum_norm/inverse.py#L761-L773)) calls `_needs_eeg_average_ref_proj` ([here](https://github.com/mne-tools/mne-python/blob/master/mne/io/proj.py#L756-L766)), which returns True if  (1) there are EEG channels present, and (2) a custom ref has not been applied, and (3) EEG average reference proj is not already there.

Instead, the user will get a `ValueError`.

In fact, it doesn't even seem to be possible any more to avoid this constraint by setting average reference to an empty list:

```python
raw = mne.io.read_raw_fif(sample_data_raw_file, verbose=False).crop(tmax=30).load_data()               
raw = raw.set_eeg_reference([])                                                                        
mne.minimum_norm.inverse._check_reference(raw)
# ValueError: EEG average reference is mandatory for inverse modeling, use set_eeg_reference method.
```

Reply:
Thanks @drammock